domain/status/service/service.go

```diff
@@ -165,11 +165,24 @@ type State interface {
 	GetModelStatusInfo(ctx context.Context) (status.ModelStatusInfo, error)
 }
 
+// ControllerState is the controller state required by this service. This is the
+// controller database, not the model state.
+type ControllerState interface {
+	// GetModelState returns the model state for the given model.
+	// It returns [modelerrors.NotFound] if the model does not exist for the given UUID.
+	GetModelState(context.Context, coremodel.UUID) (status.ModelState, error)
+
+	// GetModel returns the model for the given UUID.
+	// It returns [modelerrors.NotFound] if the model does not exist for the given UUID.
+	GetModel(context.Context, coremodel.UUID) (coremodel.Model, error)
```

Comment:
This function on the `ControllerState` interface only exists to help aid a test. It isn't needed on this interface. I would suggest that in the test the sql is just written directly to set the model's cloud credential as invalid.

I understand that this code is being ported and so the original implementation had a little more allowance to what it had access to. Not this is in a new domain we should be looking to tighten up aspects like this.